examples/cfd/external_aerodynamics/domino_sensitivity/main.py

```diff
@@ -0,0 +1,491 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This code defines a standalone distributed inference pipeline the DoMINO model.
+This inference pipeline can be used to evaluate the model given an STL and
+an inflow speed. The pre-trained model checkpoint can be specified in this script
+or inferred from the config input_file. The results are calculated on a point cloud
+sampled in the volume around the STL and on the surface of the STL. They are stored
+in a dictionary, which can be written out for visualization.
+"""
+
+from functools import cached_property
+from pathlib import Path
+import hydra
+from omegaconf import DictConfig
+from tqdm import tqdm
+
+import numpy as np
+import torch
+
+from physicsnemo.models.domino.model import DoMINO
+from physicsnemo.utils.domino.utils import unnormalize
+from torch.nn.parallel import DistributedDataParallel
+from physicsnemo.distributed import DistributedManager
+
+from numpy.typing import NDArray
+import pyvista as pv
+from .design_datapipe import DesignDatapipe
+from dataclasses import dataclass
+
+
+@dataclass
+class DoMINOInference:
+
+    cfg: DictConfig
+    model_checkpoint_path: Path | str | None = None
+    dist: DistributedManager | None = None
+    device: torch.device | None = None  # If not set, default set in __post_init__
+    model: torch.nn.Module | None = None  # If not set, constructed in __post_init__
+
+    def __post_init__(self):
+        if self.device is None:  # Sets a default device, if not specified
+            if self.dist is not None:
+                self.device = self.dist.device
+            elif torch.cuda.is_available():
+                self.device = torch.device("cuda")
+            else:
+                self.device = torch.device("cpu")
+
+        if self.model is None:
+            self.model = (
+                DoMINO(
+                    input_features=3,
+                    output_features_vol=self.num_vol_vars,
+                    output_features_surf=self.num_surf_vars,
+                    model_parameters=self.cfg.model,
+                )
+                .to(self.device)
+                .eval()
+            )
+
+            for param in self.model.parameters():
+                param.requires_grad = False
+
+            self.model = torch.compile(self.model, disable=True)  # TODO review
+
+            if self.model_checkpoint_path is not None:
+                with open(self.model_checkpoint_path, "rb") as f:
+                    self.model.load_state_dict(torch.load(f, map_location=self.device))
+                print("Model loaded with checkpoint...")
+            else:
+                print("Model loaded without checkpoint...")
```

Comment:
Should there perhaps be something more aggressive here?  Running inference without checkpoints doesn't make any sense unless you're just unit testing.  This could raise an exception, unless some unit testing flag / env variable is set?

Reply:
I see what you mean, though I'd rather raise a warning than an exception! 

There are some (though few) valid reasons to run inference without a checkpoint - in addition to unit testing, the end user might be monkey-patching weights/biases in from some other source. I also think it's good policy to allow dataclasses to be instantiated with as little validation as possible, since they're mutable - it's hard to anticipate all the ways an end-user might hack at it.

I'll implement a warning!

Reply:
Done: https://github.com/NVIDIA/physicsnemo/pull/973/commits/085077fb30d4ae70343c7023cf92fa1b9ec9516a